heim-host/src/sys/windows/users.rs

```diff
@@ -27,6 +31,44 @@ impl User {
         }))
     }
 
+    pub fn try_from_sid(sid: PSID) -> Result<Self> {
+        // name and domain cannot be longer than 256
+        let mut name_cch: DWORD = 256;
+        let mut name: Vec<WCHAR> = Vec::with_capacity(name_cch as usize);
+        let mut domain_cch: DWORD = 256;
+        let mut domain: Vec<WCHAR> = Vec::with_capacity(domain_cch as usize);
+
+        // winapi does not have SID_NAME_USE enum
+        let mut account_type: SID_NAME_USE = 0;
+
+        let result = unsafe {
+            LookupAccountSidW(
+                ptr::null(),
+                sid,
+                name.as_mut_ptr(),
+                &mut name_cch,
+                domain.as_mut_ptr(),
+                &mut domain_cch,
+                &mut account_type as PSID_NAME_USE,
+            )
+        };
+
+        if result == 0 {
+            return Err(Error::last_os_error());
```

Comment:
I would really appreciate if you add error context in here:
```suggestion
            return Err(Error::last_os_error().with_ffi("LookupAccountSidW"));
```